### DISCUSSION TITLE
missing CYRF protocols on AtMega328p target

### DISCUSSION THREAD

**Discussion by tew2012** [pascallanger/DIY-Multiprotocol-TX-Module#722] [General]
target: AtMega328p
Multi: Multiprotocol v1.3.3.14
transceiver: only CYRF6936 connected

DSMX etc. works, but the following two protocols are missing in serial mode in the OpenTX menu with the AtMega328p target, and the blink code shows in PPM mode: "The selected protocol is not available on the module". The same source code works fine on a STM32 module. Is there a way to get more build debug messages to figure out why the two protocols aren't included in the build.

#define	E010R5_CYRF6936_INO // #include "iface_rf2500.h"- not ok
#define	E129_CYRF6936_INO // #include "iface_rf2500.h" - not ok

#define	DEVO_CYRF6936_INO // "iface_cyrf6936.h" - ok
#define	DSM_CYRF6936_INO // "iface_cyrf6936.h - ok
#define	DSM_RX_CYRF6936_INO // "iface_cyrf6936.h - ok
#define	E01X_CYRF6936_INO // #include "iface_hs6200.h" - ok
#define	J6PRO_CYRF6936_INO // "iface_cyrf6936.h" - ok
#define	LOSI_CYRF6936_INO // "iface_cyrf6936.h" - ok
#define	MLINK_CYRF6936_INO // "iface_cyrf6936.h - ok
#define	TRAXXAS_CYRF6936_INO // "iface_cyrf6936.h" - ok
#define	WFLY_CYRF6936_INO // "iface_cyrf6936.h" - ok
#define	WK2x01_CYRF6936_INO // "iface_cyrf6936.h" - ok

Both protocols seems to use the RF2500 emulation.

Protocol Name|Protocol Number|Sub_Proto 0|Sub_Proto 1|Sub_Proto 2|Sub_Proto 3|Sub_Proto 4|Sub_Proto 5|Sub_Proto 6|Sub_Proto 7|RF Module|Emulation

[E010R5](Protocols_Details.md#E010R5---81)|81|||||||||CYRF6936|RF2500
[E129](Protocols_Details.md#E129---83)|83|E129|C186|||||||CYRF6936|RF2500

**Comment by pascallanger** [comment]
The RF2500 emulation does not fit on an atmega as indicated in the protocols: https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/blob/master/Protocols_Details.md#e010r5---81

**Comment by tew2012** [comment]
does the emulation need STM32 hardware or does it need more memory or other resources? I don't need all CYRF protocols, only a few if the amount of memory is the issue. And thanks for the reply. I would like to solve this.

**Reply by pascallanger** [reply]
From what I recall, the atmega did not go fast enough to read/write on the spi bus since it does not use the spi hardware but write directly to the pins instead.

**Reply by tew2012** [reply]
that's a pitty, thank you

**Reply by tew2012** [reply]
one last question. Can you tell me where those protocols get excluded (condition) in the source code? I haven't figured it out yet. Thanks

**Reply by pascallanger** [reply]
Validate.h

**Reply by tew2012** [reply]
Thanks !
